### PR TITLE
adding filtering to docworks dts

### PR DESCRIPTION
**Pull request by matanlb**

### CODE REVIEW

File: docworks-cli/src/docworks.js [hunk 1]
```diff
@@ -185,14 +188,15 @@ function docworks() {
     const outputFileName = argv.out
     const outputDirName = argv.dir || ''
     const summaryTemplate = argv.summaryTemplate
+    const ignoredServices = argv.ignore ? castArray_(argv.ignore) : []
```
Comment (agankarin) [inline docworks-cli/src/docworks.js:191]: `docworks dts -l ../wix-code-docs -i` will store `[true]`

File: docworks-cli/src/docworks.js [hunk 2]
```diff
@@ -175,6 +176,8 @@ function docworks() {
       .describe('o', 'output file')
       .alias('d', 'dir')
       .describe('d', 'output dir')
+      .alias('i', 'ignore')
+      .describe('i', 'ignores package and does not include it in the output. can be specified multiple times')
```
Comment (agankarin) [inline docworks-cli/src/docworks.js:180]: How can we pass multiple packages? (and please add a test for this case)

File: docworks-dts/lib/dts-repo.js [hunk 3]
```diff
@@ -96,6 +97,10 @@ function dts(
     }
   }
 
+  services = services.filter(service =>
```
Comment (agankarin) [inline docworks-dts/lib/dts-repo.js:100]: please add an integration test here as well

File: docworks-cli/test/dts-e2e.spec.js [hunk 4]
```diff
@@ -64,7 +65,23 @@ describe('dts workflow e2e', function () {
...
+    await runCommand('./bin/docworks dts -l ./test/docworks-service -o ./tmp/globals3'.split(' '))
+    await runCommand('./bin/docworks dts -i Service2 -l ./test/docworks-service -o ./tmp/globals4'.split(' '))
+
+    const fullContent = await fsExtra.readFile('tmp/globals3.d.ts', 'utf-8')
+    const filteredContent = await fsExtra.readFile('tmp/globals4.d.ts', 'utf-8')
+
+    const moduleRegex = /declare module '.*?'/gm
+    const allModules = fullContent.match(moduleRegex)
+    const filterModules = filteredContent.match(moduleRegex)
+    expect(filterModules.length).to.equal(allModules.length - 1)
```
Comment (agankarin) [inline docworks-cli/test/dts-e2e.spec.js:85]: I think it's worth to check that we also filter the right module, and not just the length

**Review by agankarin — CHANGES_REQUESTED** [review]

File: docworks-cli/test/dts-e2e.spec.js [hunk 5]
```diff
@@ -14,6 +14,7 @@ describe('dts workflow e2e', function () {
     const escapedString = _.escapeRegExp(string)
     const convertedWhiteSpaces = escapedString.replace(/[\s]+/g, '[\\s]+')
 
+    console.log('=========>> new RegExp(convertedWhiteSpaces)', new RegExp(convertedWhiteSpaces))
```
[the remark this answers is not in the document]
Comment (matanlb) [reply docworks-cli/test/dts-e2e.spec.js:17]: Done

File: docworks-cli/test/dts-e2e.spec.js [hunk 6]
```diff
@@ -64,7 +65,23 @@ describe('dts workflow e2e', function () {
         */
       declare module 'Service' {
           function operation(param: string): void;
-      
+
       }`))
   })
+
+  it.only('generate dts from a with ignore some services', async function () {
```
[the remark this answers is not in the document]
Comment (matanlb) [reply docworks-cli/test/dts-e2e.spec.js:72]: Done

File: docworks-cli/src/docworks.js (hunk 1 above)
Comment (matanlb) [reply docworks-cli/src/docworks.js:191]: Done

File: docworks-dts/lib/dts-repo.js [hunk 7]
```diff
@@ -96,6 +97,10 @@ function dts(
...
+    !some_(ignoredServices, ignored => service.name === ignored || service.memberOf === ignored)
```
[the remark this answers is not in the document]
Comment (matanlb) [reply docworks-dts/lib/dts-repo.js:101]: Done, separate filtering by namespaces or module name

File: docworks-cli/test/docworks-service/service.service.json [hunk 8]
```diff
@@ -1,4 +1,4 @@
-{
+[{
```
[the remark this answers is not in the document]
Comment (matanlb) [reply docworks-cli/test/docworks-service/service.service.json:1]: Done

File: docworks-cli/src/docworks.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (matanlb) [reply docworks-cli/src/docworks.js:180]: Improved documentation

Comment (matanlb) [reply docworks-cli/src/docworks.js:180]: improved documentation

File: docworks-cli/test/dts-e2e.spec.js (hunk 4 above)
Comment (matanlb) [reply docworks-cli/test/dts-e2e.spec.js:85]: Done

File: docworks-dts/lib/dts-repo.js (hunk 3 above)
Comment (matanlb) [reply docworks-dts/lib/dts-repo.js]: Done

**Pull request merged**